src/test/java/com/autotune/analyzer/utils/TestEvalExParser.java

```diff
@@ -0,0 +1,105 @@
+            "(( throughput / transaction_response_time) /  max_response_time) * 100",
+            "request_sum/request_count",
+            "(1.25 * request_count) - (1.5 * (request_sum / request_count)) - (0.25 * request_max)",
+            "((request_count / (request_sum / request_count)) / request_max) * 100"
+    ));
+    @Test
+    public void testValidate() {
+
+    for (String objFunction : objFunctionsList) {
+        ArrayList<Metric> functionVariables = getFunctionVariables();
+        assertEquals(true, new EvalExParser().validate(objFunction, functionVariables));
+        }
+    }
+
+    private ArrayList<Metric> getFunctionVariables() {
+
+        String sloJson = "{\"function_variables\":[{\"name\":\"request_sum\"," +
+                "\"query\":\"rate(http_server_requests_seconds_sum{method=\\\"GET\\\"," +
+                "outcome=\\\"SUCCESS\\\",status=\\\"200\\\",uri=\\\"/db\\\",}[1m])\"," +
+                "\"datasource\":\"prometheus\",\"value_type\":\"double\"},{\"name\":\"request_count\"," +
+                "\"query\":\"rate(http_server_requests_seconds_count{method=\\\"GET\\\"," +
+                "outcome=\\\"SUCCESS\\\",status=\\\"200\\\",uri=\\\"/db\\\",}[1m])\"," +
+                "\"datasource\":\"prometheus\",\"value_type\":\"double\"},{\"name\":\"request_max\"," +
+                "\"query\":\"http_server_requests_seconds_max{method=\\\"GET\\\",outcome=\\\"SUCCESS\\\"," +
+                "status=\\\"200\\\",uri=\\\"/db\\\"}\",\"datasource\":\"prometheus\",\"value_type\":\"double\"}]}";
+
+        JSONObject functionVariablesObject = new JSONObject(sloJson.toString());
+        JSONArray functionVariables = functionVariablesObject.getJSONArray(AnalyzerConstants.AutotuneObjectConstants.FUNCTION_VARIABLES);
+
+        ArrayList<Metric> metricArrayList = new ArrayList<>();
+
+        for (Object functionVariableObj : functionVariables) {
+            JSONObject functionVariableJson = (JSONObject) functionVariableObj;
+            String variableName = functionVariableJson.optString(AnalyzerConstants.AutotuneObjectConstants.NAME);
+            String query = functionVariableJson.optString(AnalyzerConstants.AutotuneObjectConstants.QUERY);
+            String datasource = functionVariableJson.optString(AnalyzerConstants.AutotuneObjectConstants.DATASOURCE);
+            String valueType = functionVariableJson.optString(AnalyzerConstants.AutotuneObjectConstants.VALUE_TYPE);
+
+            Metric metric = new Metric(variableName,
+                    query,
+                    datasource,
+                    valueType);
+
+            metricArrayList.add(metric);
+        }
+        return metricArrayList;
+    }
+
+    @Test
+    public void testParse() {
+
+        for(String objFunction : objFunctionsList) {
+            Map<String, String> objFunctionMap = new HashMap<>();
+            assertEquals("objective_function_map is missing or empty\n", new EvalExParser().parse(objFunction, "String", objFunctionMap));
```

Comment:
This needs to be in a error constants class

Reply:
Added